### PR TITLE
tests: using single sh snap in interface tests

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change includes:
. test-snapd-sh used on several tests
. snaps which where just doing sh are removed
. test-snapd-check-fs snap removed

### CODE REVIEW

**Comment by zyga** [comment]
I'm a bit tired today so please forgive me but I think when introducing a change that affects a large number of files it is easier to review such change if it only does one thing. Not having read this in detail yet do you think there are some parts that could be proposed separately, or at perhaps the single commit could be split into separate stages that do one thing each?

File: tests/main/interfaces-network-setup-control/task.yaml [hunk 1]
```diff
@@ -4,33 +4,31 @@ details: |
...
 systems: [-ubuntu-core-*]
 
 prepare: |
-    echo "Given the test-snapd-check-fs-access snap is installed"
-    cp "$TESTSLIB"/snaps/test-snapd-check-fs-access/meta/snap.yaml "$TESTSLIB"/snaps/test-snapd-check-fs-access/meta/snap.yaml.orig
-    sed 's/\[\]/\[network-setup-control\]/g' -i "$TESTSLIB"/snaps/test-snapd-check-fs-access/meta/snap.yaml
-    snap try "$TESTSLIB"/snaps/test-snapd-check-fs-access
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
```
Comment (bboozzoo) [inline tests/main/interfaces-network-setup-control/task.yaml]: Wow, much nicer!

File: tests/main/interfaces-network-setup-control/task.yaml [hunk 2]
```diff
@@ -4,33 +4,31 @@ details: |
...
 
 restore: |
-    cp "$TESTSLIB"/snaps/test-snapd-check-fs-access/meta/snap.yaml.orig "$TESTSLIB"/snaps/test-snapd-check-fs-access/meta/snap.yaml
+    rm -f call.error
+    rm -f /etc/netplan/test001
```
Comment (bboozzoo) [inline tests/main/interfaces-network-setup-control/task.yaml]: Are these created by the test?

File: tests/main/interfaces-network-setup-control/task.yaml [hunk 3]
```diff
@@ -4,33 +4,31 @@ details: |
...
 
     echo "When the interface is connected"
-    snap connect test-snapd-check-fs-access:network-setup-control
+    snap connect test-snapd-sh:network-setup-control
 
     echo "Then the snap is able to write in the network and netplan directories"
     for dir in $dirs; do
         if [ -d "$dir" ]; then
-            test-snapd-check-fs-access.write-dir "$dir"
+            test-snapd-sh.with-network-setup-control-plug -c "touch $dir/test001"
```
Comment (bboozzoo) [inline tests/main/interfaces-network-setup-control/task.yaml]: Ok, they are

File: tests/lib/snaps/test-snapd-check-fs-access/meta/snap.yaml [hunk 4]
```diff
@@ -1,18 +0,0 @@
-name: test-snapd-check-fs-access
```
Comment (bboozzoo) [inline tests/lib/snaps/test-snapd-check-fs-access/meta/snap.yaml]: Is this snap not used by any test?

Comment (sergiocazzolato) [reply tests/lib/snaps/test-snapd-check-fs-access/meta/snap.yaml]: I removed the uses, it was being used by 2 tests

File: tests/main/interfaces-network-setup-control/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/main/interfaces-network-setup-control/task.yaml]: yes

**Comment by sergiocazzolato** [comment]
@mvo5 failing test fixed

File: tests/main/interfaces-network-setup-control/task.yaml [hunk 5]
```diff
@@ -4,33 +4,31 @@ details: |
...
-    sed 's/\[\]/\[network-setup-control\]/g' -i "$TESTSLIB"/snaps/test-snapd-check-fs-access/meta/snap.yaml
-    snap try "$TESTSLIB"/snaps/test-snapd-check-fs-access
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
 
 restore: |
-    cp "$TESTSLIB"/snaps/test-snapd-check-fs-access/meta/snap.yaml.orig "$TESTSLIB"/snaps/test-snapd-check-fs-access/meta/snap.yaml
+    rm -f call.error
+    rm -f /etc/netplan/test001 /etc/netplan/test002
```
Comment (mvo5) [inline tests/main/interfaces-network-setup-control/task.yaml]: This line and the following line look exactly identical. Is this intentional or a copy/paste error?

**Pull request merged**